src/testUtils/TestRig.react.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+
+export var TestRigComponent = React.createClass({
+  contextTypes: {
+    triggerComplete: React.PropTypes.func.isRequired,
+    finish: React.PropTypes.func
+  },
+  getInitialState: function() {
+    return { keyVal : Math.random()};
+  },
+  componentDidUpdate: function(prevProps, prevState) {
+    this.context.finish()
+  },
+  triggerFinalRender: function() {
+    this.setState({keyVal: Math.random()})
+  },
+  render() {
+    return <div key={this.state.keyval}>{this.props.children}</div>
+  }
+})
+
+export default class TestRig {
+  constructor(TestComponent, afterRender) {
+    if (TestComponent) this.boltOn(TestComponent, afterRender)
+    if (!afterRender) this.childContext = { finish: ()=>{console.log("No expectations")} }
+  }
+
+  boltOn(TestComponent, afterRender) {
+    this.div = document.createElement('div');
+    var domNode, component
+
+    if (afterRender) this.childContext = {finish: afterRender }
```

Comment:
@BJK If `TestComponent` doesn't exist, shouldn't the `constructor` just error/finish at that point? Without a `TestComponent` the conditional `if(TestComponent)` ensures that `boltOn` is not called--which means there is no call to `React.withContext`--and therefore it is irrelevant if `this.childContext` is set at all. Unless I'm missing something, then, there is no reason to define `this.childContext` outside of `boltOn`. Instead `boltOn` should include an `if`/`else` to set `this.childContext` in all situations in which `TestComponent` exists:
```jsx
if (afterRender)
  this.childContext = {finish: afterRender }
else
  this.childContext = {finish: () => {console.log("No expectations")}}
```